classes/Tools14.php

```diff
@@ -24,2244 +24,2241 @@
-         * Delete directory and subdirectories
-         *
-         * @param string $dirname Directory name
-         */
-        public static function deleteDirectory($dirname, $delete_self = true)
-        {
-            $dirname = rtrim($dirname, DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR;
-            if (file_exists($dirname))
-                if ($files = scandir($dirname))
-                {
-                    foreach ($files as $file)
-                        if ($file != '.' && $file != '..' && $file != '.svn')
-                        {
-                            if (is_file($dirname.$file))
-                                unlink($dirname.$file);
-                            elseif (is_dir($dirname.$file.DIRECTORY_SEPARATOR))
-                                self::deleteDirectory($dirname.$file.DIRECTORY_SEPARATOR, true);
-                        }
-                    if ($delete_self && file_exists($dirname))
-                        if (!rmdir($dirname))
-                            return false;
-                    return true;
-                }
-            return false;
-        }
+    public static function safePostVars()
+    {
+        $_POST = array_map(array('Tools', 'htmlentitiesUTF8'), $_POST);
+    }
 
-        /**
-         * Display an error according to an error code
-         *
-         * @param string $string Error message
-         * @param boolean $htmlentities By default at true for parsing error message with htmlentities
-         */
-        public static function displayError($string = 'Fatal error', $htmlentities = true)
-        {
-            return $string;
-            global $_ERRORS, $cookie;
-
-            $iso = strtolower(Language::getIsoById((is_object($cookie) AND $cookie->id_lang) ? (int)$cookie->id_lang : (int)Configuration::get('PS_LANG_DEFAULT')));
-            @include_once(_PS_TRANSLATIONS_DIR_.$iso.'/errors.php');
-
-            if (defined('_PS_MODE_DEV_') AND _PS_MODE_DEV_ AND $string == 'Fatal error')
-                return ('<pre>'.print_r(debug_backtrace(), true).'</pre>');
-            if (!is_array($_ERRORS))
-                return str_replace('"', '&quot;', $string);
-            $key = md5(str_replace('\'', '\\\'', $string));
-            $str = (isset($_ERRORS) AND is_array($_ERRORS) AND key_exists($key, $_ERRORS)) ? ($htmlentities ? htmlentities($_ERRORS[$key], ENT_COMPAT, 'UTF-8') : $_ERRORS[$key]) : $string;
-            return str_replace('"', '&quot;', stripslashes($str));
-        }
+    /**
+     * Delete directory and subdirectories
+     *
+     * @param string $dirname Directory name
+     */
+    public static function deleteDirectory($dirname, $delete_self = true)
+    {
+        $dirname = rtrim($dirname, DIRECTORY_SEPARATOR).DIRECTORY_SEPARATOR;
+        if (file_exists($dirname))
+            if ($files = scandir($dirname))
+            {
+                foreach ($files as $file)
+                    if ($file != '.' && $file != '..' && $file != '.svn')

```

Comment:
This function must be cleaned or you can use Symfony FileSystem as above to delete directories :)